Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -371,50 +495,97 @@ def setDefaultParameters(self, parameterNode):
         if not parameterNode.GetParameter("Invert"):
             parameterNode.SetParameter("Invert", "false")
 
-    def process(
-        self, inputVolume, outputVolume, imageThreshold, invert=False, showResult=True
-    ):
+    def process(self, dataDirectory, patientPrefix):
         """
         Run the processing algorithm.
         Can be used without GUI widget.
-        :param inputVolume: volume to be thresholded
-        :param outputVolume: thresholding result
-        :param imageThreshold: values above/below this threshold will be set to 0
-        :param invert: if True then values above the threshold will be set to 0,
-               otherwise values below are set to 0
-        :param showResult: show output volume in slice viewers
+        :param dataDirectory: root directory containing data
+        :param patientPrefix: prefix string for selecting which files are relevant
         """
 
-        if not inputVolume or not outputVolume:
-            raise ValueError("Input or output volume is invalid")
+        if not (isinstance(dataDirectory, str) and os.path.isdir(dataDirectory)):
+            raise ValueError(
+                f"Data directory (value={repr(dataDirectory)}) is not valid"
+            )
+        if not (patientPrefix is None or isinstance(patientPrefix, str)):
+            raise ValueError(
+                f"Patient prefix (value={repr(patientPrefix)}) is not valid"
+            )
+        if patientPrefix is None:
+            patientPrefix = ""
 
         import time
 
         startTime = time.time()
         logging.info("Processing started")
 
-        # Compute the thresholded output volume using the "Threshold Scalar Volume" CLI
-        # module
-        cliParams = {
-            "InputVolume": inputVolume.GetID(),
-            "OutputVolume": outputVolume.GetID(),
-            "ThresholdValue": imageThreshold,
-            "ThresholdType": "Above" if invert else "Below",
-        }
-        cliNode = slicer.cli.run(
-            slicer.modules.thresholdscalarvolume,
-            None,
-            cliParams,
-            wait_for_completion=True,
-            update_display=showResult,
+        list_of_records = self.find_files_with_prefix(
+            dataDirectory, patientPrefix, subjectless_too=True
         )
-        # We don't need the CLI module node anymore, remove it to not clutter the scene
-        # with it
-        slicer.mrmlScene.RemoveNode(cliNode)
+        # Sort by modification time
+        list_of_records.sort(key=lambda record: record[1])
+        # Remove modification times
+        list_of_files = [record[0] for record in list_of_records]
 
         stopTime = time.time()
         logging.info(f"Processing completed in {stopTime-startTime:.2f} seconds")
 
+        return list_of_files
+
+    def find_files_with_prefix(self, path, prefix, subjectless_too=False):
```

Comment:
suggesting rename `subjectless_too` to `include_subjectless`

Reply:
Done.